capsules/src/process_console.rs

```diff
@@ -170,6 +174,68 @@ impl Default for WriterState {
     }
 }
 
+#[derive(Copy, Clone)]
+pub struct Command<const LEN: usize> {
+    buf: [u8; LEN],
+    len: usize,
+}
+impl<const LEN: usize> Command<LEN> {
+    pub fn new() -> Command<LEN> {
+        Command {
+            buf: [0; LEN],
+            len: 0,
+        }
+    }
+
+    pub fn fill(&mut self, buf: &[u8], terminator_idx: usize) -> Result<(), ErrorCode> {
```

Comment:
Please add a comment about what this function does.

Reply:
Done